packages/core/src/exchange/listToken.ts

```diff
@@ -0,0 +1,94 @@
+import type { Signer } from 'ethers'
+import { BigNumber } from 'ethers'
+import type { OfferInputBis, Sdk } from '../graphql'
+import type {
+  Address,
+  ChainId,
+  EIP712Data,
+  IAction,
+  TransactionHash,
+  UUID,
+} from '../types'
+import {
+  toAddress,
+  toAssetId,
+  toCurrencyId,
+  toTransactionHash,
+} from '../utils/convert'
+import { signEIP712 } from '../utils/signature'
+import { approveCollection } from './approveCollection'
+
+export type Action =
+  | IAction<'APPROVAL_SIGNATURE', {}>
+  | IAction<'APPROVAL_PENDING', { txHash: TransactionHash }>
+  | IAction<'OFFER_SIGNATURE', { signature: EIP712Data }>
+
+export type Listing = {
+  chain: ChainId
+  collection: Address
+  token: string
+  unitPrice: {
+    amount: BigNumber
+    currency: Address | null
```

Comment:
There is some inconsistence in term of variable naming.
I see only `takerAddress` has the prefix `Address`.
I will suggest to go with more explicit specially because we have multiple ids for some ressources.

Reply:
[c0d240a](https://github.com/liteflow-labs/liteflow-js/pull/142/commits/c0d240a3520151aafecb3feb0e9e5b74534f2334)

Variable normalized with the rest and removed the `Address` suffix.

We may need a call to discuss these variables as suffixing everything does not make sense as it's a repetition with the type, and in that case, we need consistency and suffix absolutely everything with their types, and I think that will also be interesting for the id part

Reply:
Let's go with the short version now instead of `attributeType` format, as this format is not consistent across all the attributes. With this format, we should, for example have `expiredAtDate`